File: 1018-largest-perimeter-triangle/1018-largest-perimeter-triangle.c
int compare(const void *a, const void *b) {
    return (*(int*)b - *(int*)a);
}

int largestPerimeter(int* nums, int numsSize) {
    if (numsSize < 3) return 0;
    
    qsort(nums, numsSize, sizeof(int), compare);
    
    for (int i = 0; i < numsSize - 2; i++) {
        if (nums[i] < nums[i + 1] + nums[i + 2]) {
            return nums[i] + nums[i + 1] + nums[i + 2];
        }
    }
    return 0;
}

File: 1018-largest-perimeter-triangle/1018-largest-perimeter-triangle.java
class Solution {
    public boolean isTriangle(int a, int b, int c) {
        if (a + b > c && a + c > b && b + c > a)
            return true;
        return false;
    }

    public int largestPerimeter(int[] nums) {
        int count = 0;
        Arrays.sort(nums);
        for (int i = nums.length - 1; i >= 2; i--) {
            int a = nums[i], b = nums[i - 1], c = nums[i - 2];
            if (isTriangle(a, b, c)) {
                return a + b + c;
            }
        }
        return count;

    }
}